agent/api_manager/osv_service_api.py

```diff
@@ -0,0 +1,194 @@
+            cvss_v3_vector = _get_cvss_v3_vector(vulnerability.get("severity"))
+            references = []
+            for reference in vulnerability.get("references"):
+                references.append(reference.get("url"))
+            vuln = VulnData(
+                risk=risk,
+                description=description,
+                summary=summary,
+                fixed_version=fixed_version,
+                cvss_v3_vector=cvss_v3_vector,
+                references=vulnerability.get("references"),
+                cves=filtered_cves,
+            )
+            parsed_vulns.append(vuln)
+
+        return parsed_vulns
+
+    except json.JSONDecodeError as e:
+        logger.error(f"Error decoding JSON: {e}")
+        return []
+
+
+def construct_vuln(
+    parsed_vulns: list[VulnData], package_name: str | None, package_version: str | None
+) -> Iterator[osv_output_handler.Vulnerability]:
+    """Construct Vulneravilities from the parse output.
+    Args:
+        parsed_vulns: list of VulnData.
+        package_name: The package name.
+        package_version: The package version.
+    Yields:
+        Vulnerability entry.
+    """
+    for vuln in parsed_vulns:
+        description = (
+            f"Dependency `{package_name}` with version `{package_version}`"
+            f"has a security issue.\nThe issue is identified by CVEs: `{', '.join(vuln.cves)}`."
+        )
+        recommendation = (
+            f"We recommend updating `{package_name}` to the latest available version."
+        )
+        yield osv_output_handler.Vulnerability(
+            entry=kb.Entry(
+                title=f"Use of Outdated Vulnerable Component: "
+                f"{package_name}@{package_version}: {', '.join(vuln.cves)}",
+                risk_rating=vuln.risk,
+                short_description=vuln.summary,
+                description=description,
+                references=osv_output_handler.build_references(vuln.references),
+                security_issue=True,
+                privacy_issue=False,
+                has_public_exploit=False,
+                targeted_by_malware=False,
+                targeted_by_ransomware=False,
+                targeted_by_nation_state=False,
+                recommendation=recommendation,
+            ),
+            technical_detail=f"{vuln.description} \n#### CVEs:\n {', '.join(vuln.cves)}",
+            risk_rating=RISK_RATING_MAPPING[vuln.risk],
+        )
+
+
+def _get_fixed_version(
+    affected_data: list[dict[str, list[dict[str, list[dict[str, str]]]]]],
+) -> str:
+    fixed_version = ""
+    if affected_data is not None:
+        try:
+            ranges_data: list[dict[str, list[dict[str, str]]]] = affected_data[0].get(
+                "ranges", []
+            )
+            if ranges_data:
+                events_data = ranges_data[0].get("events", [])
+                if len(events_data) > 1:
+                    fixed_version = events_data[1].get("fixed", "")
```

Comment:
How sure are you of the order here ? 
From the API docs: "While not required, it’s also recommended to keep the events array sorted according to the affected[].ranges[].type of the range."

Reply:
which is the case here : 
![image](https://github.com/Ostorlab/agent_osv/assets/144013278/c9a36434-a04c-49ee-830e-2d5fd3d4b8bc)

its the same as here :
![image](https://github.com/Ostorlab/agent_osv/assets/144013278/9ea1ed0a-6d4e-43c0-9087-fa8a39bef2cf)